server/protocols/protocols-smtp/src/main/java/org/apache/james/smtpserver/EnforceHeaderLimitationsMessageHook.java

```diff
@@ -0,0 +1,105 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.smtpserver;
+
+import java.util.Enumeration;
+
+import jakarta.mail.Header;
+
+import org.apache.commons.configuration2.Configuration;
+import org.apache.commons.configuration2.ex.ConfigurationException;
+import org.apache.james.protocols.smtp.SMTPRetCode;
+import org.apache.james.protocols.smtp.SMTPSession;
+import org.apache.james.protocols.smtp.hook.HookResult;
+import org.apache.james.protocols.smtp.hook.HookReturnCode;
+import org.apache.mailet.Mail;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class implements an SMTP hook to enforce limitations on the headers of incoming emails.
+ *
+ * It allows configuring and enforcing two types of restrictions:
+ * - A maximum number of header lines (default: 500).
+ * - A maximum total size of headers in bytes (default: 64 KB).
+ * If any of these thresholds are exceeded, the message is rejected with an SMTP error code:
+ *   <code>552 Too many header lines</code> if the number of lines exceeds the limit.
+ *   <code>552 Header size too large</code> if the total size exceeds the limit.
+ *
+ * Example XML configuration:
+ * <pre>{
+ * <handler class="org.apache.james.smtpserver.EnforceHeaderLimitationsMessageHook">
+ *     <maxLines>500</maxLines>
+ *     <maxSize>64</maxSize>
+ * </handler>
+ * }</pre>
+ *NB: The size is specified in KB, so 64 means 64 KB.
```

Comment:
Use SizeUtils in order to specify and support a unit that will need to be documented